calour/filtering.py

```diff
@@ -74,8 +74,14 @@ def downsample(exp: Experiment, field, axis=0, num_keep=None, inplace=False):
     logger.debug('downsample on field %s' % field)
     if axis == 0:
         x = exp.sample_metadata
+        error_axis_name = 'sample'
     elif axis == 1:
         x = exp.feature_metadata
+        error_axis_name = 'feature'
+
+    if field not in x:
```

Comment:
why adding this? what's wrong with just let error happen? it decreases test coverage and provides no extra benefit?

Reply:
i like it better this way :)
1. the error message is cryptic for users who don't know the code... it gives something from pandas and not clear what the problem is.
2. Also for me, the new error message lets me see the real metadata field names, so i can see my problem without printing the dataframe etc.
3. Can extend the coverage by testing the error message raised, but i don't think critical.

Reply:
alright. i am not convinced, but fine to keep it.

Reply:
:)